src/readsbml.jl

```diff
@@ -97,32 +127,37 @@ m = readSBML("my_model.xml", doc -> begin
 end)
 ```
 """
-function readSBML(
+readSBML(
     fn::String,
     sbml_conversion = document -> nothing;
     report_severities = ["Fatal", "Error"],
-)::SBML.Model
-    doc = ccall(sbml(:readSBML), VPtr, (Cstring,), fn)
-    try
-        get_error_messages(
-            doc,
-            AssertionError("Opening SBML document has reported errors"),
-            report_severities,
-        )
+)::SBML.Model = _readSBML(:readSBML, fn, sbml_conversion, report_severities)
 
-        sbml_conversion(doc)
+"""
+    readSBML(
+        str::String,
+        sbml_conversion = document -> nothing;
+        report_severities = ["Fatal", "Error"],
+    )::SBML.Model
 
-        if 0 == ccall(sbml(:SBMLDocument_isSetModel), Cint, (VPtr,), doc)
-            throw(AssertionError("SBML document contains no model"))
-        end
+Read the SBML from the string `str` and return the contained `SBML.Model`.
 
-        model = ccall(sbml(:SBMLDocument_getModel), VPtr, (VPtr,), doc)
+The `sbml_conversion` is a function that does an in-place modification of the
```

Comment:
avoid copypastes, just refer that "the other arguments work as with `readSBML`"

Reply:
Done!